Sources/FoundationEssentials/FileManager/FileManager+Files.swift

```diff
@@ -503,9 +603,47 @@ extension _FileManagerImpl {
             #endif
             return attributes
         }
+#endif
     }
     
     func attributesOfFileSystem(forPath path: String) throws -> [FileAttributeKey : Any] {
+#if os(Windows)
+        return try path.withNTPathRepresentation { pwszPath in
+            let dwLength: DWORD = GetFullPathNameW(pwszPath, 0, nil, nil)
+            guard dwLength > 0 else {
+                throw CocoaError.errorWithFilePath(path, win32: GetLastError(), reading: true)
+            }
+
+            var szVolumeName: [WCHAR] = Array<WCHAR>(repeating: 0, count: Int(dwLength + 1))
+            guard GetVolumePathNameW(pwszPath, &szVolumeName, dwLength) else {
+                throw CocoaError.errorWithFilePath(path, win32: GetLastError(), reading: true)
+            }
+
+            var liTotal: ULARGE_INTEGER = .init()
+            var liFree: ULARGE_INTEGER = .init()
+            guard GetDiskFreeSpaceExW(&szVolumeName, nil, &liTotal, &liFree) else {
+                throw CocoaError.errorWithFilePath(path, win32: GetLastError(), reading: true)
+            }
+
+            let hr: HRESULT = PathCchStripToRoot(&szVolumeName, szVolumeName.count)
+            guard hr == S_OK || hr == S_FALSE else {
+                throw CocoaError.errorWithFilePath(path, win32: DWORD(hr & 0xffff), reading: true)
+            }
+
+            var dwVolumeSerialNumber: DWORD = 0
+            guard GetVolumeInformationW(&szVolumeName, nil, 0, &dwVolumeSerialNumber, nil, nil, nil, 0) else {
+                throw CocoaError.errorWithFilePath(path, win32: GetLastError(), reading: true)
+            }
+
+            return [
+                .systemSize: UInt64(liTotal.QuadPart),
```

Comment:
Same comment here about using `_writeFileAttributePrimitive(_:as:)` and the commented out lined below